code/modules/clothing/suits/jobs.dm

```diff
@@ -104,10 +104,54 @@
 	armor = list(melee = 50, bullet = 10, laser = 25, energy = 10, bomb = 0, bio = 0, rad = 0)
 	sprite_sheets = list("Vox" = 'icons/mob/species/vox/suit.dmi')
 	siemens_coefficient = 0.7
-	
+
 /obj/item/clothing/suit/storage/det_suit/black
 	icon_state = "detective2"
 
+/obj/item/clothing/suit/storage/det_suit/technicolor
+	desc = "A 23rd-century multi-purpose trenchcoat. It's fibres are hyper-absorbent."
+	icon = 'icons/obj/clothing/coloured_detective_coats.dmi'
+	icon_state = "suit_detective_black"
+	item_state = "suit_detective_black"
+	var/suit_color
+	contained_sprite = 1
+
+/obj/item/clothing/suit/storage/det_suit/technicolor/New()
+	if(prob(5))
+		var/probably = rand(1, 8)
+		switch(probably)

```

Comment:
Use a weighted list for this instead. Something like...

var/list/colours = list("yellow" = 2, "red" = 1....etc)
var/colour = pickweight(colours)
icon_state = "suit_detective_[colour]"
item_state = "suit_detective_[colour]"

Since the colours all follow the same format, you dont even need to include that information